cluster/webhookconfigurations/usage.yaml

```diff
@@ -0,0 +1,31 @@
+---
+# Note(turkenh): It is not possible to get this generated by kubebuilder at the moment due to
```

Comment:
But that would work if you apply resources with kustomize (or with `kubectl apply -k`) right? Here we apply this object using our init container hence need the manifest in its latest shape at build time. 
Another point is, we are not following the project structure that a standard kubebuilder project has, hence that recommendation wouldn't work out of the box.
We could introduce patching with kustomize to our build process, but I don't think it would worth the effort here especially with this webhook operating on any API rather than a new one introduced by us (i.e. generated with kubebuilder).

Reply:
> We could introduce patching with kustomize to our build process, but I don't think it would worth the effort here especially with this webhook operating on any API rather than a new one introduced by us (i.e. generated with kubebuilder).

But kubebuilder annotations could be set on the hook code so that the webhook configuration gets generated, and we would probably use that if `objectSelector` option is available to be set.

The issue is that we do no preprocessing of the generated resources, before building the image. Not sure if we wanna tweak that in this PR or leave for later.

Reply:
> The issue is that we do no preprocessing of the generated resources, before building the image. Not sure if we wanna tweak that in this PR or leave for later.

I don't see any concrete problem that requires taking on this burden at the moment.